q2_types/profile_hmms/tests/test_format.py

```diff
@@ -0,0 +1,122 @@
+            self.get_data_path("hmms/dna.hmm"), "r",
+        )
+        fmt.validate()
+
+    def test_RNASingleProfileHmmFileFmt_valid(self):
+        fmt = RNASingleProfileHmmFileFmt(
+            self.get_data_path("hmms/rna.hmm"), "r"
+        )
+        fmt.validate()
+
+    def test_ProteinSingleProfileHmmFileFmt_invalid_alph(self):
+        for typ in ["rna", "dna"]:
+            fmt = ProteinSingleProfileHmmFileFmt(
+                self.get_data_path(f"hmms/{typ}.hmm"), "r",
+            )
+            with self.assertRaisesRegex(
+                ValidationError, "Found profile with alphabet "
+            ):
+                fmt.validate()
+
+    def test_DNASingleProfileHmmFileFmt_invalid_alph(self):
+        for typ in ["rna", "amino"]:
+            fmt = DNASingleProfileHmmFileFmt(
+                self.get_data_path(f"hmms/{typ}.hmm"), "r"
+            )
+            with self.assertRaisesRegex(
+                ValidationError, "Found profile with alphabet "
+            ):
+                fmt.validate()
+
+    def test_RNASingleProfileHmmFileFmt_invalid_alph(self):
+        for typ in ["dna", "amino"]:
+            fmt = RNASingleProfileHmmFileFmt(
+                self.get_data_path(f"hmms/{typ}.hmm"), "r"
+            )
+            with self.assertRaisesRegex(
+                ValidationError, "Found profile with alphabet "
+            ):
+                fmt.validate()
+
+    def test_ProteinSingleProfileHmmFileFmt_too_many_profiles(self):
+        fmt = ProteinSingleProfileHmmFileFmt(
+            self.get_data_path("hmms/4_amino.hmm"), "r"
+        )
+        with self.assertRaisesRegex(
+            ValidationError, "Expected 1 profile, found 4."
+        ):
+            fmt.validate()
+
+    def test_ProteinMultipleProfileHmmFileFmt_valid(self):
+        fmt = ProteinMultipleProfileHmmFileFmt(
+            self.get_data_path("hmms/4_amino.hmm"), "r"
+        )
+        fmt.validate()
+
+    def test_DNAMultipleProfileHmmFileFmt_valid(self):
+        fmt = DNAMultipleProfileHmmFileFmt(
+            self.get_data_path("hmms/2_dna.hmm"), "r"
+        )
+        fmt.validate()
+
+    def test_RNAMultipleProfileHmmFileFmt_valid(self):
+        fmt = RNAMultipleProfileHmmFileFmt(
+            self.get_data_path("hmms/2_rna.hmm"), "r"
+        )
+        fmt.validate()
+
+    def test_mixed_hmm_profiles_invalid_1(self):
+        fmt = ProteinMultipleProfileHmmFileFmt(
+            self.get_data_path("hmms/amino_dna.hmm"), 'r'
+        )
+        with self.assertRaisesRegex(
+            pyhmmer.errors.AlphabetMismatch, "Expected amino alphabet"
+        ):
+            fmt.validate()
+
+    def test_mixed_hmm_profiles_invalid_2(self):
+        fmt = RNAMultipleProfileHmmFileFmt(
+            self.get_data_path("hmms/rna_dna.hmm"), 'r'
+        )
+        with self.assertRaisesRegex(
+            pyhmmer.errors.AlphabetMismatch, "Expected RNA alphabet"
+        ):
+            fmt.validate()
```

Comment:
During validation we always want to raise the `ValidationError` - please catch those `AlphabetMismatch` exceptions and re-raise a `ValidationError`.